### PR TITLE
implement security scan workflow

### PR DESCRIPTION
**Pull request by tuxtof**
Implement GitHub action security scan against Blackduck and CodeQL

### CODE REVIEW

File: .github/workflows/codeql-analysis.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      # ✏️ If the Autobuild fails above, remove it and uncomment the following
+      #    three lines and modify them (or add more) to build your code if your
+      #    project uses a compiled language
+
+      #- run: |
+      #     make bootstrap
+      #     make release
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v1
```
Comment (thunderboltsid) [inline .github/workflows/codeql-analysis.yaml:52]: newline at the end of file

File: .github/workflows/codeql-analysis.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+
+name: "Code Scanning - Action"
+
+on:
+  push:
+    # branches: [main]
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: '30 1 * * 0'
```
Comment (thunderboltsid) [inline .github/workflows/codeql-analysis.yaml:10]: Consider adding a comment explaining the cron syntax e.g. 1:30 on Sundays

File: .github/workflows/synopsys.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: Black Duck Policy Check
+on:
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron:  '0 0 * * *'
```
Comment (thunderboltsid) [inline .github/workflows/synopsys.yaml:7]: Consider adding a comment explaining the cron syntax e.g. at midnight everyday

File: .github/workflows/codeql-analysis.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (thunderboltsid) [inline .github/workflows/codeql-analysis.yaml:1]: consider removing extraneous whitespace at beginning of file

File: .github/workflows/codeql-analysis.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+name: "Code Scanning - Action"
+
+on:
+  push:
+    # branches: [main]
```
Comment (thunderboltsid) [inline .github/workflows/codeql-analysis.yaml:6]: Is there a reason for leaving this commented out?

File: .github/workflows/codeql-analysis.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        # Override language selection by uncommenting this and choosing your languages
+        # with:
+        #   languages: go, javascript, csharp, python, cpp, java
```
Comment (thunderboltsid) [inline .github/workflows/codeql-analysis.yaml:33]: Do we need to set this to `go` or is the default language selection smart enough?

**Review by thunderboltsid — DISMISSED** [review]
LGTM; pending comments

**Comment by tuxtof** [comment]
@thunderboltsid comments fixed, please review

**Review by yannickstruyf3 — APPROVED** [review]

File: .github/workflows/codeql-analysis.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
+
+name: "Code Scanning - Action"
+
+on:
+  push:
+    # branches: [main]
```
[the remark this answers is not in the document]
Comment (tuxtof) [reply .github/workflows/codeql-analysis.yaml]: already done

**Review by thunderboltsid — APPROVED** [review]
LGTM

**Pull request merged**